pkg/cli/zip.go

```diff
@@ -428,7 +428,15 @@ func (zc *debugZipContext) dumpTableDataForZip(
 			}
 			// Pump the SQL rows directly into the zip writer, to avoid
 			// in-RAM buffering.
-			return sqlExecCtx.RunQueryAndFormatResults(ctx, conn, w, io.Discard, stderr, clisqlclient.MakeQuery(query))
+			return sqlExecCtx.RunQueryAndFormatResults(
+				ctx,
+				conn,
+				w,
+				io.Discard,
+				stderr,
+				clisqlclient.MakeQuery(query),
+				true, /* escapeNewline */
```

Comment:
I don't think it's reasonable to set this unconditionally.

The zip command supports a command line flag to change the format of the retrieved files IIRC. With Json for example this escaping would be incorrect.

Reply:
I changed this to only affect `csvReporter`.